cfme/tests/integration/test_cfme_auth.py

```diff
@@ -275,3 +277,68 @@ def test_login_local_group(appliance, local_users, local_group, soft_assert):
             soft_assert(local_group.description.lower() in display_groups,
                         'local group "{}" not displayed in UI groups list "{}"'
                         .format(local_group.description, display_groups))
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream('5.8')
+@pytest.mark.uncollectif(lambda auth_mode: auth_mode == 'amazon',
+                         'Amazon auth_data needed for group switch testing')
+def test_user_group_switching(appliance, auth_user_data, auth_mode, user_type, auth_provider,
+                              soft_assert, request):
+    """Test switching groups on a single user, between retreived group and built-in group"""
+    UserGroups = namedtuple("UserGroups", "user, retrieved, groups")
+    user_group_tuples = [
+        UserGroups(
+            appliance.collections.users.simple_user(
+                user.username.replace(' ', '-') if user_type == 'upn' else user.username,
+                credentials[user.password]['password'],
+                fullname=user.fullname),
+            retrieve_group(appliance, auth_mode, user.username, group, auth_provider),
+            user.groups
+        )
+        for user in auth_user_data
+        for group in user.groups
+        # pick non-evm group when there are multiple groups for the user
+        if 'evmgroup' not in group.lower() and len(user.groups) > 1]
+
+    logger.info('USER_GROUP_TUPLES: %r', user_group_tuples)
+
+    for test_user in user_group_tuples:
+        with test_user.user:
+            view = navigate_to(appliance.server, 'LoggedIn', wait_for_view=True)
+            display_name = view.current_fullname
+            display_groups = [name for name in view.group_names]
+            active_group = view.current_groupname
+            # Check there are multiple groups displayed
+            soft_assert(len(display_groups) > 1, 'Only a single group is displayed for the user')
+            display_other_group = set([g for g in display_groups if g != active_group]).pop()
+            # check the user name is displayed
+            soft_assert(display_name == test_user.user.name,
+                        'user full name "{}" did not match UI display name "{}"'
+                        .format(test_user.user, display_name))
+            # Not checking current group, determined by group priority
+            # check retreived group is there
+            soft_assert(test_user.retrieved.description in display_groups,
+                        'user group "{}" not displayed in UI groups list "{}"'
+                        .format(test_user.retrieved.description, display_groups))
+
+            # change to the other group
+            view.change_group(display_other_group)
```

Comment:
*OPTIONAL*
The other option is, if you keep the above as a soft_assert, maybe only run this portion of the test if display_other_group is not 'None' ? I am just trying to avoid the test failing with some hard-to-decipher error.